File: monte-carlo-jogo-de-moeda.R
# Estimate the prob from the frequency
obtain_relative_frequency = function(v) {
 df  = data.frame( table(v) ) 
 vx  = as.integer( as.vector(df$v) ) # ~ unique()
 fa  = as.integer( as.vector(df$Freq) ) 
 vfr = fa/sum(fa)  #table(vx)/length(vx) 
 return( data.frame(vx,vfr) )
}


moeda = c(-1,1)


# N jogadas de uma moeda
N       = 10^5
results = sample(moeda,N,replace=TRUE) 
output  = obtain_relative_frequency( results ) 


png( sprintf('moeda-N%d-from-R.png',N) ) 
title=sprintf('R: Monte Carlo simulation, N=%d',N)
xlabel=c('cara (-1) ','coroa (1)')
barplot(output$vfr, main=title,
  names.arg=xlabel, ylab="Relative Frequency")
abline(h=1/length(moeda),lty=2)
dev.off() # close

File: monte-carlo-jogo-de-moeda.py
import numpy as np
import pandas as pd
import collections as cl
from matplotlib import pylab as plt

# Estimate the prob from the frequency
def obtain_relative_frequency(v):
 count = cl.Counter( v )
 x     = np.array( list( count.keys()   ) )
 fa    = np.array( list( count.values() ) )
 fr    = 1.0*fa/np.sum(fa)
 return x,fr

moeda = np.array([-1,1])

# N jogadas de uma moeda
N       = 10**5
results = np.random.choice(moeda, size=N)
vx, vfr = obtain_relative_frequency( results )

# Plot with matplotlib 
xlabel=['cara (-1)','coroa (1)']
plt.bar(xlabel, vfr)
plt.axhline(y=1/2, color='k', linestyle='--')
plt.ylabel('Relative Frequency', fontsize=14)
plt.title('Python: Monte Carlo simulation, N=%d'%(N))
plt.savefig( 'moeda-N%d-from-python.png'%(N) )
plt.close()

